packages/game-server/src/shared/extensions/interactive.ts

```diff
@@ -0,0 +1,51 @@
+import { GenericEntity } from "../entities";
+import { Player } from "../entities/player";
+import { Extension, ExtensionNames, ExtensionSerialized } from "./types";
+
+interface InteractiveOptions {
+  eventName?: string;
+}
+
+export default class Interactive implements Extension {
+  public static readonly Name = ExtensionNames.interactive;
+
+  private self: GenericEntity;
+  private eventName: string | null = null;
+
+  public constructor(self: GenericEntity) {
+    this.self = self;
+  }
+
+  public init(options: InteractiveOptions): this {
```

Comment:
why not just put this as a second argument of the constructor?

Reply:
the reason is to not have any params, so you can either `init` or `deserialize` extension